### PR TITLE
Fix: Atomic Form Angie support > inputs are not wrapped with an atomic form container [ED-24881]

### PR DESCRIPTION
**Pull request by davseve**
## 1. Problem & Context

Form fields weren't being validated as wrapped within atomic form containers during composition building, allowing invalid structure. Adding structural validation catches orphaned fields, missing/duplicate submit buttons, and empty message elements at build time.

## 2. What Changed (Where)

- **form-structure/utils.ts**: Added three validation functions (`collectFormAncestorErrors`, `collectSubmitButtonErrors`, `collectEmptyMessageErrors`) to detect form structure violations
- **composition-builder.ts**: Integrated form validation into build pipeline, collects errors alongside existing config/style checks
- **build-composition/tool.ts**: Surfaces form errors to error handler; refactored destructuring to include new `formErrors` field
- **Test files**: Added utility helper and comprehensive test suite covering all validation scenarios

## 3. How It Works

Build process now executes three validation passes on the XML document:
1. **Form ancestor validation**: Traverses all form field elements, ensures each has `e-form` ancestor (any depth)
2. **Submit button validation**: Per-form checks that exactly one submit button exists
3. **Message validation**: Ensures success/error message elements contain children

All errors collected and returned alongside existing `configErrors`, then mapped to Error objects and reported through standard error handling pipeline.

## 4. Risks

None identified. Validation is purely additive—only detects structural problems, doesn't modify behavior for valid structures. Comprehensive test coverage (18 test cases across three functions) de-risks edge cases.

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using.
💡 **Tip:** You can customize your AI Description using **Guidelines** [Learn how](https://docs.gitstream.cm/automation-actions/#describe-changes)</sub>

### CODE REVIEW

File: packages/packages/core/editor-canvas/src/composition-builder/composition-builder.ts [hunk 1]
```diff
@@ -294,6 +295,17 @@ export class CompositionBuilder {
...
 		}
 
+		
+		const formErrors = [
+			...collectFormAncestorErrors( this.xml ),
+			...collectSubmitButtonErrors( this.xml ),
+			...collectEmptyMessageErrors( this.xml ),
+		];
+
+		if ( formErrors.length ) {
```
Comment (Ntnelbaba) [inline packages/packages/core/editor-canvas/src/composition-builder/composition-builder.ts]: This is a big failure to do, it means even one issue result to not doing anything
Make sure this is the way at this tool

File: packages/packages/core/editor-canvas/src/composition-builder/composition-builder.ts
Comment (Ntnelbaba) [inline packages/packages/core/editor-canvas/src/composition-builder/composition-builder.ts:1]: Make sure with @Omerisra6 that this fix is valid once moving it to the BE

File: packages/packages/core/editor-canvas/src/form-structure/__tests__/form-structure-utils.test.ts [hunk 2]
```diff
@@ -23,6 +26,10 @@ function mockElement( widgetType?: string, elType?: string ): V1Element {
 	} as V1Element;
 }
 
+function parseXml( xml: string ): Document {
+    return new DOMParser().parseFromString( `<root>${ xml }</root>`, 'text/xml' );
```
Comment (Ntnelbaba) [inline packages/packages/core/editor-canvas/src/form-structure/__tests__/form-structure-utils.test.ts]: Don't we have a function like that?

File: packages/packages/core/editor-canvas/src/form-structure/utils.ts [hunk 3]
```diff
@@ -115,3 +115,49 @@ export function clipboardRootsAreAtomicForms( elements: ClipboardElement[] ): bo
 
 	return elements.every( ( el ) => getClipboardElementType( el ) === FORM_ELEMENT_TYPE );
 }
+
+export function hasFormAncestor(node: Element): boolean {
```
Comment (Ntnelbaba) [inline packages/packages/core/editor-canvas/src/form-structure/utils.ts]: Wow this is so wasty
Why iterating like that? Simple search on the string can be better

File: packages/packages/core/editor-canvas/src/form-structure/utils.ts
Comment (Ntnelbaba) [inline packages/packages/core/editor-canvas/src/form-structure/utils.ts:1]: All of the file can be simplified to a better structure 
Maybe a class with constructor and simple grep if there is even a form

**Review by Ntnelbaba — CHANGES_REQUESTED** [review]
Moving this logic to BE, wait for job done

File: packages/packages/core/editor-canvas/src/form-structure/utils.ts
Comment (davseve) [reply packages/packages/core/editor-canvas/src/form-structure/utils.ts:1]: This code will be replaces in a few days.

File: packages/packages/core/editor-canvas/src/composition-builder/composition-builder.ts
Comment (davseve) [reply packages/packages/core/editor-canvas/src/composition-builder/composition-builder.ts:1]: This change will not pass to the BE implementation.
Opened a new ticket for it
https://elementor.atlassian.net/browse/ED-24891

**Review by Ntnelbaba — APPROVED** [review]
